Writer.go

```diff
@@ -24,79 +24,90 @@ import (
 )
 
 type WriterOptions struct {
-	ContentType   []byte
+	ContentTypes  [][]byte
 	Bidirectional bool
 	Timeout       time.Duration
 }
 
 type Writer struct {
-	writer *bufio.Writer
-	reader *bufio.Reader
-	opt    WriterOptions
-	buf    []byte
+	contentType []byte
+	w           *bufio.Writer
+	r           *bufio.Reader
+	opt         WriterOptions
+	buf         []byte
 }
 
-func NewWriter(w io.Writer, opt *WriterOptions) (wr *Writer, err error) {
+func NewWriter(w io.Writer, opt *WriterOptions) (writer *Writer, err error) {
 	if opt == nil {
 		opt = &WriterOptions{}
 	}
 	w = timeoutWriter(w, opt)
-	wr = &Writer{
-		writer: bufio.NewWriter(w),
-		opt:    *opt,
+	writer = &Writer{
+		w:   bufio.NewWriter(w),
+		opt: *opt,
+	}
+
+	if opt.ContentTypes != nil {
```

Comment:
```suggestion
	if len(opt.ContentTypes) > 0 {
```
otherwise can panic if non-nil but len() == 0